### PR TITLE
Support Derived Components

### PR DESCRIPTION
**Pull request by milldr**
## what
- Add support for derived components with multiple definitions of a setting

## why
- Fix broken action. If a component has the setting defined at multiple levels (defaults, base component, derived component), this action returns a list, not the last value.
- Return the last value given

## references
- [internal slack conversation for reference](https://cloudposse.slack.com/archives/C02EC1YLTV4/p1689033735557449)

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -24,5 +24,11 @@ runs:
...
       shell: bash
       run: |
-        value=$(atmos describe stacks -s ${{ inputs.stack }} --components ${{ inputs.component }} --sections=settings --format json | jq -rc --arg key "${{inputs.settings-path}}" '.. | select(.settings)?| .settings | select(has($key | split(".")[])) | getpath($key | split("."))')
+        value=$(atmos describe component \
+          ${{ inputs.component }} \
+          -s ${{ inputs.stack }} \
+          --format json | \
+          jq --arg key ${{ inputs.settings-path }} \
+          '.settings | getpath($key | split("."))' \
+        )
```
Comment (dudymas) [inline action.yml:33]: thanks for doing this 🙌

File: stacks/orgs/demo/dev/dev.yaml [hunk 2]
```diff
@@ -1,11 +1,12 @@
+# Test basic use
 components:
   terraform:
     foo:
       settings:
         level1:
           level2:
             level3:
-              secrets-arn: arn:aws:secretsmanager:us-east-1:000000000000:secret:MySecret-PlMes3
+              secrets-arn: "arn:aws:secretsmanager:us-east-1:000000000000:secret:MySecret-PlMes3"
```
Comment (dudymas) [inline stacks/orgs/demo/dev/dev.yaml:9]: just want to note that this shouldn't have a functional change, right? Unless you also wrap this in single quotes `'` like in earlier places

**Review by dudymas — APPROVED** [review]
LGTM

File: stacks/orgs/demo/dev/dev.yaml (hunk 2 above)
Comment (milldr) [reply stacks/orgs/demo/dev/dev.yaml:9]: yup I was trying to debug a failing test, but this is irrelevant. Either way translates to a string

**Pull request merged**